app/scripts/controllers/network/network-controller.test.js

```diff
@@ -50,224 +69,4106 @@ const PRE_1559_BLOCK = {
+            'NetworkController - _configureProvider - unknown type "undefined"',
+          );
+        },
+      );
+    });
+
+    for (const {
+      networkName,
+      networkType,
+      chainId,
+      networkVersion,
+    } of INFURA_NETWORKS) {
+      describe(`when the type in the provider configuration is "${networkType}"`, () => {
+        it(`initializes a provider pointed to the ${networkName} Infura network (chainId: ${chainId})`, async () => {
+          await withController(
+            {
+              state: {
+                provider: {
+                  type: networkType,
+                },
+              },
+            },
+            async ({ controller, network }) => {
+              network.mockEssentialRpcCalls();
+
+              await controller.initializeProvider();
+
+              const { provider } = controller.getProviderAndBlockTracker();
+              const promisifiedSendAsync = promisify(provider.sendAsync).bind(
+                provider,
+              );
+              const chainIdResult = await promisifiedSendAsync({
+                method: 'eth_chainId',
+              });
+              expect(chainIdResult.result).toBe(chainId);
+            },
+          );
         });
-      });
 
-      it('should set the network to loading', async () => {
-        await withController(async ({ controller }) => {
-          setupMockRpcBlockResponses({ networkType: NETWORK_TYPES.RPC });
-          setupMockRpcBlockResponses({ networkType: NETWORK_TYPES.MAINNET });
-          await controller.initializeProvider();
+        it('emits infuraIsUnblocked (assuming that the request to eth_blockNumber responds successfully)', async () => {
+          await withController(
+            {
+              state: {
+                provider: {
+                  type: 'rpc',
+                  rpcUrl: 'https://mock-rpc-url',
+                  chainId: '0x1337',
+                },
+              },
+            },
+            async ({ controller, network }) => {
+              network.mockEssentialRpcCalls();
+
+              const infuraIsUnblocked = await waitForEvent({
+                controller,
+                eventName: 'infuraIsUnblocked',
+                operation: async () => {
+                  await controller.initializeProvider();
+                },
+              });
+
+              expect(infuraIsUnblocked).toBe(true);
+            },
+          );
+        });
+
+        it(`persists "${networkVersion}" to state as the network version of ${networkName}`, async () => {
+          await withController(
+            {
+              state: {
+                provider: {
+                  type: networkType,
+                },
+              },
+            },
+            async ({ controller, network }) => {
+              network.mockEssentialRpcCalls();
+
+              await controller.initializeProvider();
+
+              expect(controller.store.getState().network).toStrictEqual(
```

Comment:
Nit: for primitive types, our lint rules require that we use `toBe` instead of `toStrictEqual`.

This rule was temporarily disabled in this repo to avoid churn, but we still intended to enable it at some point.

Reply:
Yeah I keep forgetting about this. Fixed here: [0b37b24](https://github.com/MetaMask/metamask-extension/pull/17498/commits/0b37b2452fd75097a0a25855b39e1e7de352a60d)